src/bootstrap/setup.rs

```diff
@@ -86,3 +88,41 @@ d) Install Rust from source"
     };
     Ok(template.to_owned())
 }
+
+// install a git hook to automatically run tidy --bless, if they want
+fn install_git_hook_maybe(src_path: &Path) -> io::Result<()> {
+    let mut input = String::new();
+    println!(
+        "Rust's CI will automatically fail if it doesn't pass `tidy`, the internal tool for ensuring code quality.
+If you'd like, x.py can install a git hook for you that will automatically run `tidy --bless` on each commit
+to ensure your code is up to par. If you decide later that this behavior is undesirable,
+simply delete the `pre-commit` file from .git/hooks."
+    );
+
+    let should_install = loop {
+        print!("Would you like to install the git hook?: [y/N] ");
+        io::stdout().flush()?;
+        io::stdin().read_line(&mut input)?;
```

Comment:
You need to clear `input` before reading a new line, or it will have all the previous input as well.
```
Would you like to install the git hook?: [y/N] a
error: unrecognized option 'a'
note: press Ctrl+C to exit
Would you like to install the git hook?: [y/N] y
error: unrecognized option 'a
y'
```
```suggestion
        input.clear();
        io::stdin().read_line(&mut input)?;
```

Reply:
gotcha, thanks!